data-serving/data-service/src/controllers/case.ts

```diff
@@ -75,13 +76,37 @@ export const list = async (req: Request, res: Response): Promise<void> => {
     }
 };
 
+// Finds sources with an existing URL in the request body and replace them
+// for the ones that already exist.
+// TODO: Consider calling find() with all the IDs at once if multiple sources
+// per case become the norm to avoid roundtrips to mongo.
+// TODO: Support more than just URL to fetch existing sources.
+async function fillInExistingSources(req: Request): Promise<void> {
```

Comment:
Overall thought on the approach -- do we want to just require the source ID, and match on that?

For manual and bulk entry, we can provide a search-by-name/URL dialogue that allows source creation if an appropriate one doesn't exist. For automated ingestion, we have the source ID.

Reply:
Yeah I guess I'd assumed that we'd fetch the sources in the curator portal and pass the id to the data service

Reply:
Right now the curator portal doesn't fetch any sources, the input for source is a textfield for a single URL, even if we implement a fetch->attach complete source from curator portal, having this check in here would allow us to make sure the cases we create actually has the proper denormalized sources. (Say, suggest field fails on curator portal somehow, a new source is going to be created if the case is submitted as is).
No strong opinion, I can see how it is cleaner to get rid of this PR, I can also see how things can go wrong without it :)